core/src/main/java/com/novoda/merlin/receiver/ConnectivityReceiver.java

```diff
@@ -23,28 +23,33 @@ public MerlinsBeard createMerlinsBeard(Context context) {
 
         MerlinBinderRetriever merlinBinderRetriever = new MerlinBinderRetriever() {
             @Override
-            public IBinder retrieveMerlinLocalServiceBinderIfAvailable(Context context) {
-                return peekService(context, new Intent(context, MerlinService.class));
+            public MerlinService.ConnectivityChangesNotifier retrieveConnectivityChangesNotifierIfAvailable(Context context) {
```

Comment:
Don't return an `android` specific component, in this case `IBinder`, return the business interface instead.